tests/schema.rs

```diff
@@ -1,10 +1,74 @@
+use std::collections::HashSet;
 use yang3::context::{Context, ContextFlags};
+use yang3::data::DataFormat;
 use yang3::iter::IterSchemaFlags;
 use yang3::schema::{
     DataValue, DataValueType, SchemaNodeKind, SchemaPathFormat,
 };
 
 static SEARCH_DIR: &str = "./assets/yang/";
+static YANG_LIBRARY_FILE: &str = "./assets/data/lib.json";
+
+static JSON_YANG_LIBRARY: &str = r###"
+    {
+        "ietf-yang-library:yang-library": {
+            "module-set": [
+                {
+                    "name": "complete",
+                    "module": [
+                        {
+                            "name": "iana-if-type",
+                            "revision": "2017-01-19",
+                            "namespace": "urn:ietf:params:xml:ns:yang:iana-if-type"
+                        },
+                        {
+                            "name": "ietf-interfaces",
+                            "revision": "2018-02-20",
+                            "namespace": "urn:ietf:params:xml:ns:yang:ietf-interfaces",
+                            "feature": [
+                                "arbitrary-names",
+                                "pre-provisioning",
+                                "if-mib"
+                            ]
+                        }
+                    ],
+                    "import-only-module": []
+                }
+            ],
+            "schema": [
+                {
+                    "name": "complete",
+                    "module-set": [
+                        "complete"
+                    ]
+                }
+            ],
+            "content-id": "34"
+        },
+        "ietf-yang-library:modules-state": {
+            "module-set-id": "34",
+            "module": [
+            {
+                "name": "iana-if-type",
+                "revision": "2017-01-19",
+                "namespace": "urn:ietf:params:xml:ns:yang:iana-if-type",
+                "conformance-type": "implement"
+            },
+            {
+                "name": "ietf-interfaces",
+                "revision": "2018-02-20",
+                "namespace": "urn:ietf:params:xml:ns:yang:ietf-interfaces",
+                "feature": [
+                    "arbitrary-names",
+                    "pre-provisioning",
+                    "if-mib"
+                ],
+                "conformance-type": "implement"
+                }
```

Comment:
nit: indentation seems off here. Also, please use two spaces for indentation, as in the other tests.

Reply:
Indeed. I should normalize all JSON strings to use two-space indentation in a separate commit later.